plutus-tx/src/PlutusTx/Data/List.hs

```diff
@@ -393,4 +406,187 @@ findIndex pred' = go 0 . coerce
             )
 {-# INLINEABLE findIndex #-}
 
+unzip :: forall a b . List (a, b) -> (List a, List b)
```

Comment:
Do all these have the expected complexity? If any of them are surprisingly expensive (e.g., requires more encoding/decoding than one may expect) or surprisingly cheap, please make it clear in the haddock.